daemon/config.go

```diff
@@ -501,12 +501,34 @@ func ValidateConfiguration(config *Config) error {
 	if config.IsValueSet("max-concurrent-downloads") && config.MaxConcurrentDownloads != nil && *config.MaxConcurrentDownloads < 0 {
 		return fmt.Errorf("invalid max concurrent downloads: %d", *config.MaxConcurrentDownloads)
 	}
+	if config.MaxConcurrentDownloads ==nil {
+		maxConcurrentDownloads :=defaultMaxConcurrentDownloads
+		config.MaxConcurrentDownloads = &maxConcurrentDownloads
+	}
+	if *config.MaxConcurrentDownloads < 1 {
+		//maxConcurrentDownloads :=defaultMaxConcurrentDownloads
+		//config.MaxConcurrentDownloads = &maxConcurrentDownloads
+		return fmt.Errorf("invalid max concurrent downloads: %d", *config.MaxConcurrentDownloads)
```

Comment:
if  `config.IsValueSet("max-concurrent-downloads")` return false and MaxConcurrentDownloads less than 1 , then never enter into return this error.
you can test `dockerd --max-concurrent-downloads=-1 --max-concurrent-uploads=2`,it can be normal start.